code/game/objects/items/gamemode/survivor.dm

```diff
@@ -0,0 +1,340 @@
+        return
+
+    visible_message("\The [src] emits an erroneous beep and turns off.", "You hear erroneous beep.")
+    log_game("Beacon is too far from the laptop and reset [AREACOORD(src.loc)]")
+    reset_state(FALSE)
+
+
+/obj/item/beacon/rescue/proc/reset_state(dump_contents = TRUE)
+    activated = FALSE
+    anchored = FALSE
+    icon_state = "motion0"
+    w_class = WEIGHT_CLASS_BULKY
+    SetLuminosity(initial(luminosity))
+    update_icon()
+
+    if (issurvivorgamemode(SSticker.mode))
+        var/datum/game_mode/survivor/GM = SSticker.mode
+        GM.beacon = null
+    
+    if (dump_contents && length(internal_components))
+        required_components = initial(required_components)
+        for (var/I in internal_components)
+            var/turf/T = get_step(src, pick(cardinal))
+            if (T.obscured)
+                usr.transferItemToLoc(I, src)
+            else
+                usr.transferItemToLoc(I, T)
+
+    if (beacon_timer_id && noise_timer_id)
+        distress_timer = timeleft(beacon_timer_id)
+        deltimer(noise_timer_id)
+        deltimer(beacon_timer_id)
+        for (var/mob/M in GLOB.alive_human_list)
+            SEND_SOUND(M, sound('sound/misc/notice2.ogg'))
+            to_chat(M, "<h2 class='alert'>MESSAGE RECIEVED</h2>")
+            to_chat(M, "<span class='alert'>We have lost signal with your beacon! Get it set back up or we'll never find you.</span>")
+        log_game("[src] was stopped. Timer had [distress_timer] time left.")
+
+    STOP_PROCESSING(SSobj, src)
+
+
+/obj/item/beacon/rescue/attack_alien(mob/living/carbon/Xenomorph/M)
+    if(current_hp <= 0)
+        to_chat(M, "<span class='warning'>It's already broken.</span>")
+        return
+
+    M.animation_attack_on(src)
+    M.visible_message("<span class='danger'>\The [M] slices [src] apart!</span>", \
+        "<span class='danger'>You slice [src] apart!</span>", null, 5)
+    playsound(loc, "alien_claw_metal", 25, 1)
+
+    if (activated == FALSE)
+        return
+
+    current_hp -= max(0, rand(15, 30))
+    if(current_hp <= 0)
+        log_game("[key_name(M)] destroyed \the [src].")
+        reset_state()
+   
+    if(prob(10))
+        new /obj/effect/decal/cleanable/blood/oil(src.loc)
+    
+
+// TODO: Handle other click types (altclick etc)
+/obj/item/beacon/rescue/attack_hand(mob/user as mob)
+    if(!anchored)
+        return ..()
+
+    if(!ishuman(user)) 
+        return ..()
+
+    user.visible_message("<span class='notice'>[user] starts packing up \the [src].</span>",
+    "<span class='notice'>You start packing up \the [src]. <b>This will interrupt the process.</b></span>")
+    if(!user.get_active_held_item() && do_after(user, activation_time*1.2, TRUE, 5, BUSY_ICON_FRIENDLY,, TRUE))
+        log_game("[key_name(user)] picked up \the [src].")
+        reset_state()
+        user.put_in_hands(src)
+    return ..()
+
+
+/obj/item/beacon/rescue/attackby(obj/item/W as obj, mob/user as mob)
```

Comment:
Make these use the proper parameters and give it proper inheritance logic
```suggestion
/obj/item/beacon/rescue/attackby(obj/item/I, mob/user, params)
```

Reply:
Is the `as X` bad?

Reply:
It just does nothing but clutters the code (as only works in either input as a special construct or in verbs)